### PR TITLE
Make octo commands lowercase: `Octo -> octo`

### PR DESCRIPTION
**Pull request by thedewi**
For easier invocation on case-sensitive systems (linux), and consistency with examples in documentation.

### CODE REVIEW

File: BuildAssets/octo [hunk 1]
```diff
@@ -5,7 +5,7 @@ while [ -h "$SOURCE" ]; do # resolve $SOURCE until the file is no longer a symli
   SOURCE="$(readlink "$SOURCE")"
   [[ $SOURCE != /* ]] && SOURCE="$DIR/$SOURCE" # if $SOURCE was a relative symlink, we need to resolve it relative to the path where the symlink file was located
 done
-OCTO_PATH="$( dirname "$SOURCE")/Octo.Dll"
+OCTO_PATH="$( dirname "$SOURCE")/octo.Dll"
```
Comment (matt-richardson) [inline BuildAssets/octo]: should `.Dll` be `.dll`?

File: BuildAssets/OctopusTools.nuspec [hunk 2]
```diff
@@ -2,15 +2,15 @@
 <package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd" xmlns:octopus="http://octopus.org">
   <metadata>
     <id>OctopusTools</id>
-    <title>Octo.exe</title>
+    <title>octo.exe</title>
```
Comment (matt-richardson) [inline BuildAssets/OctopusTools.nuspec]: Should this be OctopusCli?

Comment (matt-richardson) [reply BuildAssets/OctopusTools.nuspec]: This probably ties in with something I'm going to need for xplat stuff...
the [E2E Tests](https://github.com/OctopusDeploy/OctopusDeploy/blob/master/source/Octopus.E2ETests.API/Octopus.E2ETests.API.csproj#L43) currently include octo.exe... I'm going to need the xplat version as well, to run the tests on non-windows platforms.

File: readme.md [hunk 3]
```diff
@@ -1,9 +1,9 @@
-This repository contains the command line tool (`Octo.exe`) for [Octopus Deploy][1], an automated deployment server for professional .NET developers. You can use it to create and deploy releases, create and push packages, and manage environments.
+This repository contains the command line tool (`octo.exe`) for [Octopus Deploy][1], an automated deployment server for professional .NET developers. You can use it to create and deploy releases, create and push packages, and manage environments.
```
Comment (matt-richardson) [inline readme.md]: This should get the rename treatment. `Octopus CLI` (or whatever it should be). Also on line 3 and 6.

While we're at it, get @robpearson's or @MJRichardson's input on rewording `an automated deployment server for professional .NET developers`.

File: source/Octopus.Cli/CliProgram.cs [hunk 4]
```diff
@@ -154,7 +154,7 @@ static int PrintError(Exception ex)
                 Log.Error(ex.Message);
                 if (LogExtensions.IsKnownEnvironment())
                 {
-                    Log.Error("This error is most likely occurring while executing Octo.exe as part of an automated build process. The following doc is recommended to get some tips on how to troubleshoot this: https://g.octopushq.com/OctoexeTroubleshooting");
+                    Log.Error("This error is most likely occurring while executing octo.exe as part of an automated build process. The following doc is recommended to get some tips on how to troubleshoot this: https://g.octopushq.com/OctoexeTroubleshooting");
```
Comment (matt-richardson) [inline source/Octopus.Cli/CliProgram.cs]: Can we figure out what the executing assembly's name is?

Comment (matt-richardson) [reply source/Octopus.Cli/CliProgram.cs]: Will also need to create a new octopurl for that link that doesn't include `octoexe`

File: readme.md (hunk 3 above)
Comment (thedewi) [reply readme.md]: Okay, done. Negotiated a refreshed one-line description with @robpearson and I've also applied it to the OctopusDeploy repo's description and `readme` opening sentence.

File: source/Octopus.Cli/CliProgram.cs (hunk 4 above)
Comment (thedewi) [reply source/Octopus.Cli/CliProgram.cs]: Good idea. Fixed 6 instances.

Comment (thedewi) [reply source/Octopus.Cli/CliProgram.cs]: I've created a card for updating docs, and then we can do a pass over octopurls and other references to docs.

File: Dockerfiles/Readme.md [hunk 5]
```diff
@@ -1,5 +1,5 @@
 # octopusdeploy/octo
-A docker wrapped version of the popular [octo.exe](https://octopus.com/docs/api-and-integration/octo.exe-command-line) command line tool
+A docker wrapped version of the popular Octopus CLI, [octo](https://octopus.com/docs/api-and-integration/octo.exe-command-line)
```
Comment (matt-richardson) [inline Dockerfiles/Readme.md]: Note for future us to fix up that doc location

Comment (matt-richardson) [reply Dockerfiles/Readme.md]: added to https://trello.com/c/DySZV2WR/31-move-octo-docs-to-octopus-rest-api-octopus-cli-and-update-generator

File: source/Octopus.Cli/CliProgram.cs [hunk 6]
```diff
@@ -154,7 +155,7 @@ static int PrintError(Exception ex)
                 Log.Error(ex.Message);
                 if (LogExtensions.IsKnownEnvironment())
                 {
-                    Log.Error("This error is most likely occurring while executing Octo.exe as part of an automated build process. The following doc is recommended to get some tips on how to troubleshoot this: https://g.octopushq.com/OctoexeTroubleshooting");
+                    Log.Error($"This error is most likely occurring while executing {AssemblyExtensions.GetExecutableName()} as part of an automated build process. The following doc is recommended to get some tips on how to troubleshoot this: https://g.octopushq.com/OctoexeTroubleshooting");
```
Comment (matt-richardson) [inline source/Octopus.Cli/CliProgram.cs]: :heart:

**Pull request merged**